fiat-roles/src/main/java/com/netflix/spinnaker/fiat/roles/UserRolesSyncer.java

```diff
@@ -49,41 +48,44 @@
 @Slf4j
 @Component
 @ConditionalOnExpression("${fiat.writeMode.enabled:true}")
-public class UserRolesSyncer implements RunnableAgent {
+public class UserRolesSyncer {
+  private final LockManager lockManager;
+  private final PermissionsRepository permissionsRepository;
+  private final PermissionsResolver permissionsResolver;
+  private final ResourceProvider<ServiceAccount> serviceAccountProvider;
+  private final ResourceProvidersHealthIndicator healthIndicator;
 
-  @Getter
-  private final String agentType = "UserRoleSyncer";
-  @Getter
-  private final String providerName = "Fiat";
+  private final long retryIntervalMs;
+  private final long syncDelayMs;
+  private final long syncDelayTimeoutMs;
 
   @Autowired
-  @Setter
-  private PermissionsRepository permissionsRepository;
-
-  @Autowired
-  @Setter
-  private PermissionsResolver permissionsResolver;
-
-  @Autowired(required = false)
-  @Setter
-  private ResourceProvider<ServiceAccount> serviceAccountProvider;
-
-  @Autowired
-  @Setter
-  private ResourceProvidersHealthIndicator healthIndicator;
-
-  @Value("${fiat.writeMode.retryIntervalMs:10000}")
-  @Setter
-  private long retryIntervalMs = 10000;
-
-  @Value("${fiat.writeMode.syncDelayTimeoutMs:30000}")
-  @Setter
-  private long syncDelayTimeoutMs = 30000;
-
+  public UserRolesSyncer(LockManager lockManager,
+                         PermissionsRepository permissionsRepository,
+                         PermissionsResolver permissionsResolver,
+                         ResourceProvider<ServiceAccount> serviceAccountProvider,
+                         ResourceProvidersHealthIndicator healthIndicator,
+                         @Value("${fiat.writeMode.retryIntervalMs:10000}") long retryIntervalMs,
+                         @Value("${fiat.writeMode.syncDelayMs:600000}") long syncDelayMs,
+                         @Value("${fiat.writeMode.syncDelayTimeoutMs:30000}") long syncDelayTimeoutMs) {
+    this.lockManager = lockManager;
+    this.permissionsRepository = permissionsRepository;
+    this.permissionsResolver = permissionsResolver;
+    this.serviceAccountProvider = serviceAccountProvider;
+    this.healthIndicator = healthIndicator;
+
+    this.retryIntervalMs = retryIntervalMs;
+    this.syncDelayMs = syncDelayMs;
+    this.syncDelayTimeoutMs = syncDelayTimeoutMs;
+  }
 
+  @Scheduled(fixedDelayString = "${fiat.writeMode.syncDelayMs:600000}")
+  public void schedule() {
```

Comment:
I need to circle around and actually have this obey discovery status for start/stop